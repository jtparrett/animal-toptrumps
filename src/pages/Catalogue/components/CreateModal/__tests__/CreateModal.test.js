import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { ThemeProvider, theme } from "@chakra-ui/core";

import { CreateModal } from "../";

describe("<CreateModal />", () => {
  it("renders", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <MockedProvider>
          <CreateModal />
        </MockedProvider>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("opens the modal", () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <MockedProvider>
          <CreateModal />
        </MockedProvider>
      </ThemeProvider>
    );

    fireEvent.click(getByText("Create animal"));

    expect(document.body).toMatchSnapshot();
  });
});
