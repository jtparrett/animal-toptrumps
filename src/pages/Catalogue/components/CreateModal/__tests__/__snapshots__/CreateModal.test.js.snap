// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateModal /> opens the modal 1`] = `
<body>
  <div
    aria-hidden="true"
    class="Toaster"
    data-aria-hidden="true"
    id="react-toast"
  >
    <span
      class="Toaster__manager-top"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; margin: 0px auto; text-align: center; top: 0px; right: 0px; left: 0px;"
    />
    <span
      class="Toaster__manager-top-left"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; top: 0px; left: 0px;"
    />
    <span
      class="Toaster__manager-top-right"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; top: 0px; right: 0px;"
    />
    <span
      class="Toaster__manager-bottom-left"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; bottom: 0px; left: 0px;"
    />
    <span
      class="Toaster__manager-bottom"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; margin: 0px auto; text-align: center; bottom: 0px; right: 0px; left: 0px;"
    />
    <span
      class="Toaster__manager-bottom-right"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; bottom: 0px; right: 0px;"
    />
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      class="css-c4fa1y"
      type="button"
    >
      Create animal
      <svg
        aria-hidden="true"
        class="css-yyruks"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    id="chakra-portal-3"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="css-1a29t1v"
      />
      <div
        class="css-19ewlhq"
      >
        <form
          aria-describedby="modal-3-body"
          aria-labelledby="modal-3-header"
          aria-modal="true"
          class="css-bs1nqz"
          id="modal-3"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="css-j7qwjs"
          >
            <div
              class="css-acwcvw"
              role="group"
            >
              <label
                class="css-176s5fq"
                for="name"
              >
                Name
              </label>
              <input
                class="css-1ga4kqw"
                id="name"
                name="name"
                value=""
              />
            </div>
            <div
              class="css-acwcvw"
              role="group"
            >
              <label
                class="css-176s5fq"
                for="type"
              >
                Type
              </label>
              <div
                class="css-1jke4yk"
              >
                <select
                  class="css-16ut77p"
                  id="type"
                  name="type"
                >
                  <option
                    value=""
                  >
                    Select type
                  </option>
                  <option
                    value="MAMMAL"
                  >
                    Mammal
                  </option>
                  <option
                    value="REPTILE"
                  >
                    Reptile
                  </option>
                  <option
                    value="FISH"
                  >
                    Fish
                  </option>
                  <option
                    value="AMPHIBIOUS"
                  >
                    Amphibious
                  </option>
                </select>
                <div
                  class="css-1lepn2w"
                >
                  <svg
                    aria-hidden="true"
                    class="css-12c4avn"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="css-acwcvw"
              role="group"
            >
              <label
                class="css-176s5fq"
                for="diet"
              >
                Dietry requirements
              </label>
              <div
                class="css-1jke4yk"
              >
                <select
                  class="css-16ut77p"
                  id="diet"
                  name="diet"
                >
                  <option
                    value=""
                  >
                    Select diet type
                  </option>
                  <option
                    value="CARNIVORE"
                  >
                    Carnivore
                  </option>
                  <option
                    value="HERBIVORE"
                  >
                    Herbivore
                  </option>
                </select>
                <div
                  class="css-1lepn2w"
                >
                  <svg
                    aria-hidden="true"
                    class="css-12c4avn"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="css-0"
              role="group"
            >
              <div
                class="css-70qvj9"
              >
                <label
                  class="css-1eqsc50"
                  for="extinct"
                >
                  Extinct
                </label>
                <label
                  class="css-v7ibj3"
                >
                  <input
                    class="css-1gy22ay"
                    id="extinct"
                    name="extinct"
                    type="checkbox"
                    value=""
                  />
                  <div
                    aria-hidden="true"
                    class="css-yb72im"
                  >
                    <div
                      class="css-qyllxa"
                    />
                  </div>
                </label>
              </div>
            </div>
          </div>
          <div
            class="css-yx2vxv"
          >
            <button
              aria-disabled="false"
              class="css-1as6f6t"
              type="submit"
            >
              Create animal
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<CreateModal /> renders 1`] = `
<DocumentFragment>
  <button
    class="css-c4fa1y"
    type="button"
  >
    Create animal
    <svg
      aria-hidden="true"
      class="css-yyruks"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
        fill="currentColor"
      />
    </svg>
  </button>
</DocumentFragment>
`;
